modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/HttpHeadersAuthenticatorUtilsTests.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.http.netty4;
+
+import io.netty.handler.codec.http.DefaultHttpRequest;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpVersion;
+
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.http.netty4.HttpHeadersAuthenticatorUtils.HttpHeadersWithAuthenticationContext;
+import org.elasticsearch.test.ESTestCase;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+public final class HttpHeadersAuthenticatorUtilsTests extends ESTestCase {
```

Comment:
should there be a sanity test to wrap/extract and unwrap ?

Reply:
I can do that, but this should mostly be covered by code in `Netty4HttpServerTransportTests#testHttpHeadersSuccessfulValidation`.
The authn context is wrapped with a request type that is then processed by netty into a different request type, that we then unwrap. The code doing the processing from one request type to another is more accessible in `*HttpServerTransportTests` type of tests rather than utility test classes here.

Reply:
If it is covered elsewhere that's good enough the question was just more about coverage than needing specific tests here.

Reply:
The 2 new tests in `Netty4HttpServerTransportTests` require a lot of setup. For this reason, each test asserts multiple aspects of the validation flow, that I would normally prefer to have covered in separate tests. I resisted it because I reckon this way the overall readability would decrease...

However in [d143282](https://github.com/elastic/elasticsearch/pull/95112/commits/d143282dc08b85db5375d6baf1dbdb050687e1cd) I have pushed a new test that asserts a properly wrapped request can be unwrapped.